ledger-core/v2/logicrunner/sm_request/call_result.go

```diff
@@ -45,22 +45,15 @@ func (s *StateMachineCallResult) stepProcess(ctx smachine.ExecutionContext) smac
 	outgoingRef := reference.NewGlobal(*s.Payload.Caller.GetLocal(), s.Payload.CallOutgoing)
 	link, bgin := ctx.GetPublishedGlobalAliasAndBargeIn(outgoingRef)
 	if link.IsZero() {
-		// TODO: log
-		return ctx.Stop()
+		return ctx.Error(throw.E("no one is waiting", struct{ smachine.SlotLink }{link}))
 	}
 	if bgin == nil {
-		// TODO: log
-		return ctx.Stop()
-	}
-	if !bgin.IsValid() {
-		// TODO: log
-		return ctx.Stop()
+		return ctx.Error(throw.E("impossible situation ", struct{ smachine.BargeInHolder }{bgin}))
```

Comment:
Replace with simple `ctx.Error(throw.Impossible())`, bgin is nil, so we don't need no info here